File: Ruby/Loot.rb
#encoding: UTF-8

# Autores: Manuel Díaz-Meco Terrés y Ana Graciani Donaire
# 2ºDGIIM		PDOO


# Esta clase representa el botín que se obtiene al vencer a una nave enemiga. Puede incluir 
# cantidades que representen un número de paquetes de suministros, armas, potenciadores de
# de escudo, hangares y/o medallas.

require_relative 'LootToUI.rb'

module Deepspace
	class Loot

		# Constructor

		def initialize (supplies, weapons, shields, hangars, medals, efficient = false, spacecity = false)
			# Atributos de instancia (son privados)
			@nSupplies = supplies
			@nWeapons = weapons
			@nShields = shields
			@nHangars = hangars
			@nMedals = medals
			@getEfficient = efficient
			@spaceCity = spacecity
		end

		public

		# Consultores

		def nSupplies
			return @nSupplies
		end

		def nWeapons
			return @nWeapons
		end

		def nShields
			return @nShields
		end

		def nHangars
			return @nHangars
		end

		def nMedals
			return @nMedals
		end
		
		def getEfficient
			return @getEfficient
		end
		
		def spaceCity
			return @spaceCity
		end	
			
		def to_s
			return getUIversion.to_s
		end
		
		def getUIversion
			return LootToUI.new(self)
		end

	end
	
	#Comprobación de código:
=begin
	l = Loot.new(2, 3, 4, 5, 6)
	puts l.inspect
	puts l.nSupplies
	puts l.nWeapons
	puts l.nShields
	puts l.nHangars
	puts l.nMedals
	puts l.to_s
=end
end

File: Java/deepspace/src/deepspace/Loot.java
/**
 * @authors Manuel Diaz-Meco Terrés y Ana Graciani Donaire
 * 2ºDGIIM         PDOO
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 * Esta clase representa el botín que se obtiene al vencer a una nave enemiga
 * Puede incluir cantidades representen un número de paquetes de suministros,
 * armas, potenciadores de escudo, hangares y/o medallas.
 */
public class Loot {
    
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
    
    private boolean getEfficient = false;
    private boolean spaceCity = false;

    /**
     * Constructor con 5 parámetros, no hay transformaciones
     * @param nSuppgiven
     * @param nWgiven
     * @param nShigiven
     * @param nHgiven
     * @param nMgiven
     * @param ef
     * @param city 
     */
    Loot (int nSuppgiven, int nWgiven, int nShigiven, int nHgiven, int nMgiven) {
        
        nSupplies = nSuppgiven;
        nWeapons = nWgiven;
        nShields = nShigiven;
        nHangars = nHgiven;
        nMedals = nMgiven;
    }

    
    /**
     * Constructor con 7 parámetros, puede haber transformaciones
     * @param nSuppgiven
     * @param nWgiven
     * @param nShigiven
     * @param nHgiven
     * @param nMgiven
     * @param ef
     * @param city 
     */
    Loot (int nSuppgiven, int nWgiven, int nShigiven, int nHgiven, int nMgiven,
          boolean ef, boolean city) {
        
        nSupplies = nSuppgiven;
        nWeapons = nWgiven;
        nShields = nShigiven;
        nHangars = nHgiven;
        nMedals = nMgiven;
        getEfficient = ef;
        spaceCity = city;
    }
    
    // Consultor público para el atributo nSupplies
    
    public int getNSupplies() { return nSupplies; }
    
    // Consultor público para el atributo nWeapons          
    
    public int getNWeapons() { return nWeapons; }
    
    // Consultor público para el atributo nShields
    
    public int getNShields() { return nShields; }
    
    // Constructor público para el atributo nHangars
    
    public int getNHangars() { return nHangars; }
    
    // Consultor público para el atributo nMedals
    
    public int getNMedals() { return nMedals; }
    
    // Consultor público para el atributo efficient
    
    public boolean getEfficient() { return getEfficient; }
    
    // Consultor público para el atributo spaceCity
    
    public boolean spaceCity() { return spaceCity; }
    
    /**
     * Método toString
     * @return String que representa el objeto Loot
     */
    @Override
    public String toString() {
        return "nSupplies: " + nSupplies + ", nWeapons: " + nWeapons +
                ", nShields: " + nShields + ", nHangars: " + nHangars +
                ", nMedals: " + nMedals + ", efficient: " + getEfficient +
                ", spaceCity: " + spaceCity;
    }
    
        LootToUI getUIversion() {
        return (new LootToUI(this));
    }
}
